.github/container/Dockerfile.t5x.arm64

```diff
@@ -0,0 +1,212 @@
+
+RUN python -c 'import sys; vi = sys.version_info; print(f"{vi.major}.{vi.minor}")'  > /mealkit-python-version
+
+
+#------------------------------------------------------------------------------
+# build array_record 0.5.0 from source
+#------------------------------------------------------------------------------
+# TODO: Remove this once array_record maintainers have published an arm64 wheel.
+
+FROM linaro/tensorflow-arm64-build:2.12-multipython as array_record-builder
+
+COPY --from=wheel-builder /mealkit-python-version /mealkit-python-version
+
+# The bazel build of array_record makes some strong assumptions on where the cpp toolchain lives.
+# We therefore replicate (part of) its Dockerfile, which ensures everything is in the right place:
+#   https://github.com/google/array_record/tree/v0.5.0/oss/build.Dockerfile.aarch64
+#
+# BEGIN
+    ENV DEBIAN_FRONTEND=noninteractive
+    ENV PYTHON_BIN_PATH=/usr/bin/python${MEALKIT_PYTHON_VERSION}
+
+    # Install supplementary Python interpreters
+    RUN export PYTHON_BIN_PATH=/usr/bin/python$(cat /mealkit-python-version) && \
+        ln -s ${PYTHON_BIN_PATH} /usr/local/bin/python && \
+        ln -s ${PYTHON_BIN_PATH} /usr/local/bin/python3 && \
+        ln -s ${PYTHON_BIN_PATH} /usr/bin/python
+
+    RUN apt update && \
+        apt install -yqq \
+            apt-utils \
+            build-essential \
+            checkinstall \
+            libffi-dev
+
+    # Install pip dependencies needed for array_record
+    RUN /usr/bin/python -m pip install -U pip && \
+        /usr/bin/python -m pip install -U \
+            absl-py \
+            auditwheel \
+            etils[epath] \
+            patchelf \
+            setuptools \
+            twine \
+            wheel
+# END
+
+RUN <<"EOT" bash -exu
+set -o pipefail
+
+git clone https://github.com/google/array_record.git /tmp/array_record
+cd /tmp/array_record
+git checkout v0.5.0
+
+export CROSSTOOL_TOP="@ml2014_aarch64_config_aarch64//crosstool:toolchain"
+export AUDITWHEEL_PLATFORM="manylinux2014_aarch64"
+./oss/build_whl.sh
+EOT
+
+
+#------------------------------------------------------------------------------
+# build grain from a manually-patched version
+#------------------------------------------------------------------------------
+# TODO: Remove this once grain maintainers have published an arm64 wheel.
+
+FROM wheel-builder as grain-builder
+
+# Setup bazel
+RUN  wget https://github.com/bazelbuild/bazelisk/releases/download/v1.17.0/bazelisk-linux-arm64 -O /usr/bin/bazel && \
+    chmod a+x /usr/bin/bazel
+
+# The following setup steps are based on
+#   https://github.com/google/grain/blob/ab557c31672cc2ac6f9b31e33aea2df00a7da5bf/grain/oss/build.Dockerfile
+#
+# BEGIN
+    # Setup python
+    RUN export MEALKIT_PYTHON_VERSION=$(cat /mealkit-python-version) && \
+        apt-get update && apt-get install -y \
+            python3-dev python3-pip python3-venv && \
+            rm -rf /var/lib/apt/lists/* && \
```

Comment:
Any particular reason you have this in some places but not all?
I don't think it matters as these are just in the wheel-building steps, which we ultimately don't use any layers from.

Reply:
This just replicates a step in https://github.com/google/grain/blob/ab557c31672cc2ac6f9b31e33aea2df00a7da5bf/grain/oss/build.Dockerfile#L34-L39 -- I wouldn't be surprised if we could tweak/minimize this given what's already installed in the mealkit base image, but out of convenience for staying in sync with the grain repo I decided to include it here.